src/main/java/net/pms/dlna/DLNAProtocolInfo.java

```diff
@@ -0,0 +1,265 @@
+	 */
+	public DLNAProtocolInfo(ProtocolInfo template) {
+		this(template.getProtocol(),
+			template.getNetwork(),
+			template.getContentFormat(),
+			template.getAdditionalInfo()
+		);
+	}
+
+	/**
+	 * @return The {@link DLNAOrgFlags} for this {@link DLNAProtocolInfo}.
+	 */
+	public DLNAOrgFlags getFlags() {
+		return flags != null ? flags : DLNAOrgFlags.IMPLIED;
+	}
+
+	/**
+	 * @return The {@code DLNA.ORG_PN} (DLNA media format profile) for this
+	 *         {@link DLNAProtocolInfo} or {@code null} if none is defined.
+	 */
+	public String getPN() {
+		return pn;
+	}
+
+	/**
+	 * Creates a new {@link MimeType} from the content format of this
+	 * {@link DLNAProtocolInfo}.
+	 *
+	 * @return The corresponding {@link MimeType}.
+	 * @throws ParseException If {@code contentFormat} can't be parsed to a
+	 *             {@link MimeType}.
+	 */
+	public MimeType getMimeType() throws ParseException {
+		return MimeType.valueOf(contentFormat);
+	}
+
+	/**
+	 * Creates a new {@link org.seamless.util.MimeType} from the content format
+	 * of this {@link DLNAProtocolInfo}. To create a {@link MimeType}, use
+	 * {@link #getMimeType()} instead.
+	 *
+	 * @return The corresponding {@link org.seamless.util.MimeType}.
+	 * @see #getMimeType()
+	 */
+	@Override
+	public org.seamless.util.MimeType getContentFormatMimeType() throws IllegalArgumentException {
+		return org.seamless.util.MimeType.valueOf(contentFormat);
+	}
+
+	/**
+	 * Parses {@code additionalInfo} and stores the results.
+	 */
+	@Override
+	protected void parseAdditionalInfo() {
+		if (isBlank(additionalInfo)) {
+			return;
+		}
+		String[] parameters = additionalInfo.trim().toUpperCase(Locale.ROOT).split("\\s*;\\s*");
+		for (String parameter : parameters) {
+			if (isBlank(parameter)) {
+				continue;
+			}
+			String[] parameterEntry = parameter.split("\\s*=\\s*");
+			if (parameterEntry.length == 2) {
+				DLNAAttribute.Type type =
+					DLNAAttribute.Type.valueOfAttributeName(parameterEntry[0]);
+				if (type != null) {
+					DLNAAttribute<?> dlnaAttrinute;
+					try {
+						dlnaAttrinute = DLNAAttribute.newInstance(type, parameterEntry[1], getMimeType().toStringWithoutParameters());
+					} catch (ParseException e) {
+						dlnaAttrinute = DLNAAttribute.newInstance(type, parameterEntry[1], null);
+						LOGGER.debug("Error parsing mime-type while parsing additionalInfo from \"{}\": {}", this, e.getMessage());
+						LOGGER.trace("", e);
+					}
+					if (dlnaAttrinute != null) {
+						attributes.put(type, dlnaAttrinute);
+					}
+					switch (type) {
+						case DLNA_ORG_FLAGS:
+							flags = new DLNAOrgFlags(parameterEntry[1]);
+							break;
+						case DLNA_ORG_PN:
+							pn = parameterEntry[1];
+							break;
+						case DLNA_ORG_CI:
+						case DLNA_ORG_OP:
+						case DLNA_ORG_PS:
+						default:
+							break;
+					}
+				} else if (!"SONY.COM_PN".equals(parameterEntry[0])) {
+					LOGGER.debug("Unknown ProtocolInfo parameter type \"{}\" ({})", parameterEntry[0], parameter);
```

Comment:
@Nadahar I stupidely thought that it was a "standard" operation.
As Java is my first object language experience, i'm surprise.
You're right that it need to be "rational" to keep the full precision for the PS flag, avoiding desynchronisation over time.
Though i don't know what you mean when saying there are bad at rounding ? if it's less than 1/1000 that should be fine/suffisant for our use ?

That's doesn't mean that you need to make it perfect or the more general possible...i don't even understand why, for what other UMS use ?